# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/public/xfig/gcc40.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur xfig.3.2.4/u_fonts.c xfig.3.2.4-p/u_fonts.c
--- xfig.3.2.4/u_fonts.c	2002-09-09 19:36:58.000000000 +0200
+++ xfig.3.2.4-p/u_fonts.c	2005-10-21 00:52:34.000000000 +0200
@@ -21,7 +21,7 @@
 
 /* X11 font names */
 
-struct _xfstruct x_fontinfo[NUM_FONTS] = {
+struct _xfstruct _x_fontinfo[NUM_FONTS] = {
     {"-*-times-medium-r-normal--", (struct xfont*) NULL},
     {"-*-times-medium-i-normal--", (struct xfont*) NULL},
     {"-*-times-bold-r-normal--", (struct xfont*) NULL},
@@ -58,6 +58,7 @@
     {"-*-itc zapf chancery-medium-i-normal--", (struct xfont*) NULL},
     {"-*-itc zapf dingbats-*-*-*--", (struct xfont*) NULL},
 };
+struct _xfstruct *x_fontinfo = _x_fontinfo;
 
 /* Use the following font names for any font that doesn't exist in the table above.
  * These come with the Open Group X distribution so they should be a common set.
@@ -66,7 +67,7 @@
  * structure when needed by draw_text() to scale text down below MIN_FONT_SIZE points.
 */
 
-struct _xfstruct x_backup_fontinfo[NUM_FONTS] = {
+struct _xfstruct _x_backup_fontinfo[NUM_FONTS] = {
     {"-*-times-medium-r-normal--", (struct xfont*) NULL},
     {"-*-times-medium-i-normal--", (struct xfont*) NULL},
     {"-*-times-bold-r-normal--", (struct xfont*) NULL},
@@ -103,10 +104,11 @@
     {"-*-zapf chancery-medium-i-normal--", (struct xfont*) NULL},
     {"-*-zapf dingbats-*-*-*--", (struct xfont*) NULL},
 };
+struct _xfstruct *x_backup_fontinfo = _x_backup_fontinfo;
 
 /* PostScript font names matched with X11 font names in x_fontinfo */
 
-struct _fstruct ps_fontinfo[NUM_FONTS + 1] = {
+struct _fstruct _ps_fontinfo[NUM_FONTS + 1] = {
     {"Default", -1},
     {"Times-Roman",			0},
     {"Times-Italic",			1},
@@ -144,10 +146,11 @@
     {"ZapfChancery-MediumItalic",	33},
     {"ZapfDingbats",			34},
 };
+struct _fstruct *ps_fontinfo = _ps_fontinfo;
 
 /* LaTeX font names and the corresponding PostScript font index into ps_fontinfo */
 
-struct _fstruct latex_fontinfo[NUM_LATEX_FONTS] = {
+struct _fstruct _latex_fontinfo[NUM_LATEX_FONTS] = {
     {"Default",		0},
     {"Roman",		0},
     {"Bold",		2},
@@ -155,6 +158,7 @@
     {"Sans Serif",	16},
     {"Typewriter",	12},
 };
+struct _fstruct *latex_fontinfo = _latex_fontinfo;
 
 x_fontnum(psflag, fnum)
     int		    psflag, fnum;
diff -dur xfig.3.2.4/u_fonts.h xfig.3.2.4-p/u_fonts.h
--- xfig.3.2.4/u_fonts.h	2002-03-27 01:18:23.000000000 +0100
+++ xfig.3.2.4-p/u_fonts.h	2005-10-21 00:49:24.000000000 +0200
@@ -32,9 +32,9 @@
 
 extern int		psfontnum();
 extern int		latexfontnum();
-extern struct _xfstruct	x_fontinfo[], x_backup_fontinfo[];
-extern struct _fstruct	ps_fontinfo[];
-extern struct _fstruct	latex_fontinfo[];
+extern struct _xfstruct	*x_fontinfo, *x_backup_fontinfo;
+extern struct _fstruct	*ps_fontinfo;
+extern struct _fstruct	*latex_fontinfo;
 
 /* element of linked list for each font
    The head of list is for the different font NAMES,
diff -dur xfig.3.2.4/w_i18n.c xfig.3.2.4-p/w_i18n.c
--- xfig.3.2.4/w_i18n.c	2002-11-08 23:18:16.000000000 +0100
+++ xfig.3.2.4-p/w_i18n.c	2005-10-21 00:55:57.000000000 +0200
@@ -402,7 +402,7 @@
      XFontSet *fontset;
      int *size_ret;
 {
-  extern struct _xfstruct x_fontinfo[]; /* X11 fontnames */
+  extern struct _xfstruct *x_fontinfo; /* X11 fontnames */
   static int font_types[] = { 0, 2 };  /* 0 = Times-Roman, 2 = Times-Bold */
   struct xfont *nf;
   int i;
